packages/components-react/src/examples/CheckboxExampleControlled.tsx

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+import {
+  PCheckbox,
+  PText,
+  type CheckboxUpdateEventDetail,
+  type PCheckboxProps,
+} from '@porsche-design-system/components-react';
+
+export const CheckboxControlledExamplePage = (): JSX.Element => {
+  const [value, setValue] = useState<PCheckboxProps['value']>('');
+
+  const onUpdate = (e: CustomEvent<CheckboxUpdateEventDetail>) => {
+    setValue(e.detail.checked ? e.detail.value : '');
+  };
+
+  return (
+    <>
+      <PCheckbox
+        label="Some Label"
+        name="some-name"
+        value="some-value"
+        onUpdate={(e) => onUpdate(e as CustomEvent<CheckboxUpdateEventDetail>)}
+      />
+      <PText>Value: {value}</PText>
+    </>
+  );
+};
```

Comment:
```suggestion
export const CheckboxControlledExamplePage = (): JSX.Element => {
  const [state, setState] = useState({
    'some-name': true
  });

  const onUpdate = (e: CustomEvent<CheckboxUpdateEventDetail>) => {
    setState((prev) => ({
      ...prev,
      [e.detail.value]: e.detail.checked,
    }));
  };

  return (
    <>
      <PCheckbox
        label="Some Label"
        name="some-name"
        checked={state['some-name']}
        value="some-value"
        onUpdate={(e) => onUpdate(e as CustomEvent<CheckboxUpdateEventDetail>)}
      />
      <PText>Value: {value}</PText>
    </>
  );
};
```

Realistically in a controlled approach you would store the checked state together with the name in an external state and then update the checked state accordingly. Please change for all controlled examples.